Client/Frontend/Browser/HomePanel/FavoritesViewController.swift

```diff
@@ -216,6 +310,71 @@ class FavoritesViewController: UIViewController, Themeable {
         collection.collectionViewLayout.invalidateLayout()
     }
     
+    private func setupBackgroundImage() {
+        guard var background = backgroundImage.info,
+            let image = background.image else {
+            return
+        }
+        
+        let imageAspectRatio = image.size.width / image.size.height
+        let imageView = UIImageView(image: image)
+        
+        imageView.contentMode = UIImageView.ContentMode.scaleAspectFit
+        view.addSubview(imageView)
+        
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.snp.makeConstraints {
+            
+            // Determines the height of the content
+            // `999` priority is required for landscape, since top/bottom constraints no longer the most important
+            //   using `high` is not enough either.
+            $0.top.bottom.equalToSuperview().priority(ConstraintPriority(999))
+            
+            // In portrait `top`/`bottom` is enough, however, when switching to landscape, those constraints
+            //  don't force centering, so this is used as a stronger constraint to center in landscape/portrait
+            $0.centerY.equalToSuperview()
+            
+            // Width of the image view is determined by the forced height constraint and the literal image ratio
+            $0.width.equalTo(imageView.snp.height).multipliedBy(imageAspectRatio)
+            
+            // These are required constraints to avoid a bad center pushing the image out of view.
+            // if a center of `-100` or `100000` is specified, these override to keep entire background covered by image.
+            // The left side cannot exceed `0` (or superview's left side), otherwise whitespace will be shown on left.
+            $0.left.lessThanOrEqualToSuperview()
+            
+            // the right side cannot drop under `width` (or superview's right side), otherwise whitespace will be shown on right.
+            $0.right.greaterThanOrEqualToSuperview()
+            
+            // If for some reason the image cannot fill full width (e.g. not a landscape image), then these constraints
+            //  will fail. A constraint will be broken, since cannot keep both left and right side's pinned
+            //  (due to the width multiplier being < 1
+            
+            // Using `high` priority so that it will not be applied / broken  if out-of-bounds.
+            // Offset updated / calculated during view layout as views are not setup yet.
+            let backgroundConstraint = $0.left.equalTo(view.snp.centerX).priority(ConstraintPriority.high).constraint
+            self.backgroundViewInfo = (imageView, backgroundConstraint)
+        }
+    }
+    
+    fileprivate func gradientOverlay() -> CAGradientLayer {
```

Comment:
This gradient looks average on ip8, lot of white-on-white elements

<img width="544" alt="Zrzut ekranu 2019-11-26 o 13 53 17" src="https://user-images.githubusercontent.com/6950387/69635204-31fa3580-1054-11ea-80f0-2aea76622645.png">

Reply:
Yeah, kind of a gap in the spec, but didn't want to dynamically handle all of the font color variations for iOS 12 and 13, as it starts becoming a ton of combinations.